tests/ImageSharp.Tests/Image/ImageTests.WrapMemory.cs

```diff
@@ -413,6 +413,51 @@ public void WrapMemory_IMemoryOwnerOfT_ValidSize(int size, int height, int width
                 Image.WrapMemory(memory, height, width);
             }
 
+            [Theory]
+            [InlineData(0, 5, 5)]
+            [InlineData(20, 5, 5)]
+            [InlineData(1023, 32, 32)]
+            public void WrapMemory_IMemoryOwnerOfByte_InvalidSize(int size, int height, int width)
+            {
+                var array = new byte[size * Unsafe.SizeOf<Rgba32>()];
+                var memory = new TestMemoryOwner<byte> { Memory = array };
+
+                Assert.Throws<ArgumentException>(() => Image.WrapMemory<Rgba32>(memory, height, width));
+            }
+
+            [Theory]
+            [InlineData(25, 5, 5)]
+            [InlineData(26, 5, 5)]
+            [InlineData(2, 1, 1)]
+            [InlineData(1024, 32, 32)]
+            [InlineData(2048, 32, 32)]
+            public void WrapMemory_IMemoryOwnerOfByte_ValidSize(int size, int height, int width)
+            {
+                var random = new Random();
+                var pixelSize = Unsafe.SizeOf<Rgba32>();
+                var array = new byte[size * pixelSize];
+                var memory = new TestMemoryOwner<byte> { Memory = array };
+
+                using (var img = Image.WrapMemory<Rgba32>(memory, width, height))
+                {
+                    Assert.Equal(width, img.Width);
+                    Assert.Equal(height, img.Height);
+
+                    for (int i = 0; i < height; ++i)
+                    {
+                        var arrayIndex = pixelSize * width * i;
+                        var expected = (byte)random.Next(0, 256);
+
+                        Span<Rgba32> rowSpan = img.GetPixelRowSpan(i);
+                        array[arrayIndex] = expected;
+
+                        Assert.Equal(expected, rowSpan[0].R);
```

Comment:
Nit: it'd be more correct to just compare the reference directly (it'd also avoid possible false positives).
Something like:
```suggestion
                        Span<Rgba32> rowSpan = img.GetPixelRowSpan(i);
                        ref Rgba32 r0 = ref rowSpan[0];
                        ref Rgba32 r1 = ref Unsafe.As<byte, Rgba32>(ref array[arrayIndex]);
                        
                        Assert.IsTrue(Unsafe.AreSame(ref r0, ref r1));
```

You'd also need to remove that `var random = new Random()` at line 436 🙂

Reply:
Thanks! I started down this path, but couldn't figure out how to compare the memory locations of the ref vars (`Unsafe.AreSame`).

Reply:
Ahah glad I could help and thanks for making the change!
And yeah, the `Unsafe` class has a lot of great and forbidden stuff, it's lots of fun 😈